File: FlatboxEditor.Render/Logger.cs
using FlatboxEditor.FFI;

namespace FlatboxEditor.Render;

public static class Logger
{
    static Logger()
    {
        NativeInterface.logger_init();
    }

    public static void Info(string msg) {
        NativeInterface.logger_info(msg);
    }

    public static void Warn(string msg) {
        NativeInterface.logger_warn(msg);
    }

    public static void Error(string msg) {
        NativeInterface.logger_error(msg);
    }

    public static void Debug(string msg) {
        NativeInterface.logger_debug(msg);
    }
}

File: FlatboxEditor.FFI/src/logger.rs
use std::ffi::c_char;
use flatbox_core::logger::{FlatboxLogger, warn, info, error, debug};
use native_macro::native;

pub struct Logger;

#[native]
impl Logger {
    pub fn init() {
        FlatboxLogger::init();
        debug!("Logger::init()");
    }

    pub fn info(msg: &str) {
        info!("{msg}");
    }

    pub fn warn(msg: &str) {
        warn!("{msg}");
    }

    pub fn error(msg: &str) {
        error!("{msg}");
    }

    pub fn debug(msg: &str) {
        debug!("{msg}");
    }
}